apps/voting/contracts/Voting.sol

```diff
@@ -418,4 +588,32 @@ contract Voting is IForwarder, AragonApp {
         uint256 computedPct = _value.mul(PCT_BASE) / _total;
         return computedPct > _pct;
     }
+
+    function _storeVote(Vote storage vote_, uint256 _voteId, bool _supports, address _voter) private {
+        uint256 voterStake = token.balanceOfAt(_voter, vote_.snapshotBlock);
+        VoterState state = vote_.voters[_voter];
+
+        // If voter had previously voted, decrease count
+        if (state == VoterState.Yea) {
+            vote_.yea = vote_.yea.sub(voterStake);
+        } else if (state == VoterState.Nay) {
+            vote_.nay = vote_.nay.sub(voterStake);
+        }
+
+        if (_supports) {
+            vote_.yea = vote_.yea.add(voterStake);
+        } else {
+            vote_.nay = vote_.nay.add(voterStake);
+        }
+
+        vote_.voters[_voter] = _supports ? VoterState.Yea : VoterState.Nay;
+        emit CastVote(_voteId, _voter, _supports, voterStake);
+    }
+
+    function _overwriteIssuerIfNecessary(Vote storage vote_, address _voter) private {
+        address _currentIssuer = vote_.issuers[_voter];
+        if (_currentIssuer != address(0) && _currentIssuer != _voter) {
```

Comment:
Second part of the condition is in case a voter votes in behalf of himself, right?

Reply:
exactly :)

Reply:
Why don't we set it to `address(0)` then to get the gas refund if this is the case?

Reply:
Makes sense 😅